Tests/BSONTests/BSONTests.swift

```diff
@@ -19,6 +19,15 @@ open class BSONTestCase: XCTestCase {
         }
         return path
     }
+
+    struct TestError: Error {
+        let message: String
+    }
+
+    func fail(_ message: String = "This should not happen") -> Never {
```

Comment:
I don't think we would ever want to use this. `fatalError` will crash the entire test runner and prevent future tests from compelting.